### DISCUSSION TITLE
Automated Unattended Uninstall?

### DISCUSSION THREAD

**Discussion by FrankAllotta** [Homebrew/discussions#3454] [Everyday usage] [answered]
Is there a way to uninstall homebrew in an automated way? For example, the EC2 Mac instances come pre-installed with brew. However, some organizations forbid brew to be installed. I would want a way to automate the uninstallation of brew when creating an AMI with something like packer. The issue that I see that might arise is that you are prompted for a password during uninstall. How would you bypass that?

**Comment by SMillerDev** [comment, marked as answer]
The same way you do on install. `SUDO_ASKPASS`

**Reply by FrankAllotta** [reply]
I've never done it on install either. You have an example?

**Reply by SMillerDev** [reply]
I'm sure that if you search for `SUDO_ASKPASS` online you can find some examples since it's just part of how sudo works.

**Reply by FrankAllotta** [reply]
As it turns out, the user that is logged in upon AMI creation doesn't require the sudo password.  I tested it by using a packer shell provisioner and called a script that just runs the uninstall command. I just had to pass the command the 'y' to make sure it skips that interactive prompt. It uninstalled successfully when the provisioner ran the command. 🙂

**Comment by gromgit** [comment]
https://stackoverflow.com/questions/12608293/how-to-setup-a-sudo-askpass-environment-variable
